testing/osde2e/gpu-addon.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+if ! [[ -d  $ARTIFACT_DIR ]] ; then
+    echo "FATAL: osde2e tests requires '/test-run-results' to exist"
+    exit 1
+fi
+
+BURN_RUNTIME_SEC=600
+
+JUNIT_HEADER_TEMPLATE='<?xml version="1.0" encoding="utf-8"?>
+<testsuite errors="NUM_ERRORS" failures="NUM_ERRORS" name="TEST_TARGET" tests="1" time="RUNTIME" timestamp="TIMESTAMP">
+    <testcase name="TEST_TARGET" time="RUNTIME">
+        <CASE_OUTPUT_TAG>
+'
+
+JUNIT_FOOTER_TEMPLATE='
+        </CASE_OUTPUT_TAG>
+    </testcase>
+</testsuite>
+'
+
+function exit_and_abort() {
+    echo "Failed. Aborting."
+    exit 1
+}
+
+function run_test() {
+    TARGET=${1-}
+    echo "====== Running toolbox '$TARGET'"
+    TARGET_NAME=$(echo $TARGET | sed 's/ /_/g')
+    JUNIT_FILE="${ARTIFACT_DIR}/junit_${TARGET_NAME}.xml"
+
+    echo $JUNIT_HEADER_TEMPLATE > "${JUNIT_FILE}"
+
+    RAW_OUTPUT=$(/usr/bin/time -o ${ARTIFACT_DIR}/runtime  ./run_toolbox.py ${TARGET})
+    STATUS=$?
+
+    OUTPUT=$(echo $RAW_OUTPUT | jq -sRr @uri)
```

Comment:
I think this may explode any time, if the output of test gets too long.
This syntax would prevent it:
```
jq -sRr @uri <<EOF
$RAW_OUTPUT
EOF
```

Reply:
Found a better way. Using a file. and replacing angle brackets so XML wont break. Now we have better output. No need to URI encode.